security/pkg/nodeagent/test/cert_rotation/rotation_test.go

```diff
@@ -0,0 +1,48 @@
+// Copyright 2020 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package rotatesds
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"istio.io/istio/mixer/test/client/env"
+	testID "istio.io/istio/mixer/test/client/env"
+	sdsTest "istio.io/istio/security/pkg/nodeagent/test"
+)
+
+func TestCertRotation(t *testing.T) {
+	rotateInterval := 1 * time.Second
+	sdsTest.RotateCert(rotateInterval)
+	setup := sdsTest.SetupTest(t, testID.SDSCertRotation)
+	defer setup.TearDown()
+
+	setup.StartProxy(t)
+	start := time.Now()
+	for {
+		code, _, err := env.HTTPGet(fmt.Sprintf("http://localhost:%d/echo", setup.OutboundListenerPort))
+		if err != nil {
+			t.Errorf("Failed in request: %v", err)
+		}
+		if code != 200 {
+			t.Errorf("Unexpected status code: %d", code)
+		}
+		if time.Since(start) > 2*rotateInterval {
+			break
+		}
```

Comment:
we have no way of verifying the cert has actually rotated properly in this test. If you removed the rotate command above, or broke the rotation code, this test would still pass, right?

Reply:
Good point. I should add verification for that.

Reply:
@myidpt I made a change to the bootstrap config to let every request transfer over a new SSL connection. I also add stats verification.